src/Framework/Models/EagerLoader.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Give\Framework\Models;
+
+use ReflectionClass;
+
+/**
+ * @unreleased
+ */
+class EagerLoader
+{
+    /**
+     * @unreleased
+     * @var string
+     */
+    protected $modelClass;
+
+    /**
+     * @unreleased
+     * @var string
+     */
+    protected $relationshipKey;
+
+    /**
+     * @unreleased
+     * @var ModelQueryBuilder
+     */
+    protected $modelQuery;
+
+    /**
+     * @unreleased
+     * @var ModelQueryBuilder
+     */
+    protected $eagerLoadedQuery;
+
+    /**
+     * @unreleased
+     * @var string
+     */
+    protected $foreignKey;
+
+    /**
+     * @var mixed
+     */
+    protected $foreignAttribute;
+
+    public function __construct($modelClass, $relationshipKey, $eagerLoadedModelClass, $foreignKey, $foreignAttribute = null)
+    {
+        $this->modelClass = $modelClass;
+        $this->modelQuery = $modelClass::query();
```

Comment:
I think we might need to make sure the model has the `query()` method in some way as it comes from interfaces like `ModelCrud` and `ModelReadOnly`

Reply:
I decided to check against the `Model::class` as a sub-class.

Reply:
Resolved, see https://github.com/impress-org/givewp/pull/7227/commits/69707995eddb9f276cc69aa9c51272f72cc207e4.